Go Segmentation violation when network is down

## Description

CLI output when the network is down in the local machine is hard to understand.

## Steps to reproduce

1. Disconnect internet
2. Try `chor app list`

Output:
```
11:30:40 ➜  choreo-0.0.3-SNAPSHOT ./bin/chor app list                
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x4db71d]

goroutine 1 [running]:
fmt.Fprintln(0x0, 0x0, 0xc00012b668, 0x1, 0x1, 0x3, 0x97bf0a, 0xd)
	/opt/hostedtoolcache/go/1.13.3/x64/src/fmt/print.go:265 +0x5d
github.com/wso2/choreo-cli/internal/pkg/cmd/common.Println(...)
	/home/runner/work/choreo-cli/choreo-cli/internal/pkg/cmd/common/util.go:143
github.com/wso2/choreo-cli/internal/pkg/cmd/common.PrintErrorMessage(...)
	/home/runner/work/choreo-cli/choreo-cli/internal/pkg/cmd/common/util.go:36
github.com/wso2/choreo-cli/internal/pkg/client.(*cliClient).makeHttpCall(0xc0000cc7d0, 0x97bf0a, 0xd, 0x9789b2, 0x3, 0x0, 0x0, 0x8b8b40, 0x33, 0x18)
	/home/runner/work/choreo-cli/choreo-cli/internal/pkg/client/client.go:116 +0x351
github.com/wso2/choreo-cli/internal/pkg/client.(*cliClient).getRestResource(0xc0000cc7d0, 0x97bf0a, 0xd, 0x8a8820, 0xc0000aece0, 0x0, 0x0)
	/home/runner/work/choreo-cli/choreo-cli/internal/pkg/client/client.go:57 +0x7f
github.com/wso2/choreo-cli/internal/pkg/client.(*cliClient).ListApps(0xc0000cc7d0, 0xa365a0, 0xc0000cc7d0, 0x7f4270499040, 0xc0000d0780, 0x0)
	/home/runner/work/choreo-cli/choreo-cli/internal/pkg/client/application_client.go:21 +0x6f
github.com/wso2/choreo-cli/internal/pkg/cmd/application.runListAppCommand.func1(0xc000121b80, 0xd962c0, 0x0, 0x0)
	/home/runner/work/choreo-cli/choreo-cli/internal/pkg/cmd/application/list.go:42 +0x98
github.com/spf13/cobra.(*Command).execute(0xc000121b80, 0xd962c0, 0x0, 0x0, 0xc000121b80, 0xd962c0)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0xc000120780, 0xc0000a20d8, 0x1, 0x1)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
	/home/runner/work/choreo-cli/choreo-cli/cmd/choreo/choreo.go:68 +0x225
```

## Affected version

 Version:		0.0.3-SNAPSHOT
 Git commit:		a8dad5f1a043ead7a064bb54a78a01492209329d
 Built:			2019-10-30T14:26:36+0000
 OS/Arch:		linux/amd64


## Suggested solution

Ideally, the CLI should return an error like "Cannot communicate with server"

--- a/cmd/choreo/choreo.go
+++ b/cmd/choreo/choreo.go
@@ -23,10 +23,10 @@ import (
 	"github.com/wso2/choreo-cli/internal/pkg/config"
 )
 
-type NoWriter struct {
+type noWriter struct {
 }
 
-func (c *NoWriter) Write(p []byte) (int, error) {
+func (c *noWriter) Write(p []byte) (int, error) {
 	//do nothing
 	return 0, nil
 }
@@ -34,8 +34,16 @@ func (c *NoWriter) Write(p []byte) (int, error) {
 type CliContextData struct {
 	userConfig    runtime.UserConfig
 	envConfig     runtime.EnvConfig
-	verboseWriter io.Writer
+	verboseWriter *ioWriterWrapper
 	apiClient  runtime.Client
+}
+
+type ioWriterWrapper struct {
+	writer io.Writer
+}
+
+func (w *ioWriterWrapper) Write(data []byte) (n int, err error) {
+	return w.writer.Write(data)
 }
 
 func (c *CliContextData) Client() runtime.Client {
@@ -58,7 +66,9 @@ func (c *CliContextData) EnvConfig() runtime.EnvConfig {
 }
 
 func main() {
-	cliContext := &CliContextData{}
+	cliContext := &CliContextData{
+		verboseWriter: &ioWriterWrapper{writer: &noWriter{}},
+	}
 
 	initConfig(cliContext)
 	initClient(cliContext)
@@ -110,9 +120,9 @@ func cobraOnInit(cliContext *CliContextData, command *cobra.Command) func() {
 			common.ExitWithError(cliContext.Out(), "Error retrieving verbose flag value", err)
 		}
 		if verbose {
-			cliContext.verboseWriter = os.Stdout
+			cliContext.verboseWriter.writer = os.Stdout
 		} else {
-			cliContext.verboseWriter = &NoWriter{}
+			cliContext.verboseWriter.writer = &noWriter{}
 		}
 	}
 }

--- a/internal/pkg/client/client.go
+++ b/internal/pkg/client/client.go
@@ -13,6 +13,7 @@ import (
 	"bytes"
 	"encoding/json"
 	"errors"
+	"fmt"
 	"io"
 	"io/ioutil"
 	"net/http"
@@ -106,15 +107,15 @@ func (c *cliClient) createRestResource(resourcePath string, data interface{}) er
 func (c *cliClient) makeHttpCall(resourcePath string, method string, dataReader io.Reader) (*http.Response, error) {
 	req, err := NewRequest(c.backendUrl, c.accessToken, method, resourcePath, dataReader)
 	if err != nil {
-		common.PrintError(c.debug, "Error creating post request. Reason: ", err)
-		return nil, nil
+		common.PrintErrorMessage(c.debug, err.Error())
+		return nil, fmt.Errorf("error creating server request")
 	}
 
 	httpClient := NewClient(c.skipVerify)
 	resp, err := httpClient.Do(req)
 	if err != nil {
 		common.PrintErrorMessage(c.debug, err.Error())
-		return nil, nil
+		return nil, fmt.Errorf("error communicating with the server")
 	}
 
 	return resp, nil

